### DISCUSSION TITLE
RT flag A/B in json

### DISCUSSION THREAD

**Discussion by andimik** [windytan/redsea#105] [Ideas]
I realized that there is no json output for the Radiotext A/B flag.

From EN 50067

> An important feature of type 2 groups is the Text A/B flag contained in the second block. Two cases occur:
> 
> - If the receiver detects a change in the flag (from binary "0" to binary "1" or vice-versa), then the whole RadioText display should be cleared and the newly received RadioText message segments should be written into the display.
> - If the receiver detects no change in the flag, then the received text segments or characters should be written into the existing displayed message and those segments or characters for which no update is received should be left unchanged.

**Comment by windytan** [comment]
Thanks, could be a nice addition

**Comment by andimik** [comment]
`radiotext_plus. item_toggle` is existing BTW, but this is RT+.

**Comment by windytan** [comment]
The reason why a/b has been left out is it's mainly used for reconstructing or clearing the string, kind of background stuff. Similarly, things like the radiotext character address are omitted from output. But do you think you'd like to see the a/b flag as a string ("a", "b"), integer (0, 1), or bool (false, true)?

**Comment by windytan** [comment]
Here's a simple line to show it as string, it should be added between lines 532 and 533 in src/groups.cc:

```cpp
json_["radiotext_ab"] = getBits<1>(group.get(BLOCK2), 4) == 0 ? "a" : "b";
```

**Comment by windytan** [comment]
Another idea: We could make the `-p` option a general verbose option. What I mean is, if the user specifices `-p` then all kinds of background information will be displayed, like current RadioText character address, partially received strings, A/B flags, and all the RadioText+ flags as well. Without `-p` it would be more quiet, and focused on information that is (mostly) supposed to be seen by end users.

**Comment by windytan** [comment]
It's in the newest snapshot - `rt_ab` will be enabled with the `--show-partial` option
